File: 1. Two Sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        map<int, int> data;
        map<int,int>::iterator it;

        int cur;
        for (int i = 0; i < nums.size(); i++){
            int cur = nums[i];
            if (data.empty()){
                data.insert({cur, i});
            } else {
                it = data.find(target - cur);
                if (it == data.end()) {
                    data.insert({cur, i});
                } else {
                    result.push_back(it->second);
                    result.push_back(i);
                    break;
                }
            }
        }

        return result;
    }
};

File: 1. Two Sum/solution.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int[] result = new int[2];
        Dictionary<int, int> data = new Dictionary<int, int>();

        int cur;
        Nullable<int> pair;
        for (int i = 0; i < nums.Length; i++){
            cur = nums[i];
            pair = (data.TryGetValue(target-cur, out int value)) ? value : null;
            if (pair != null){
                result[0] = data[target-cur];
                result[1] = i;
                break;
            } else {
                data[cur] = i;
            }
        }      
        return result;
    }
}